fizzbuzz/src/main/java/fizz.buzz/number/factory/AbstractNumberFactory.java

```diff
@@ -0,0 +1,37 @@
+package fizz.buzz.number.factory;
+
+import fizz.buzz.number.Number;
+import fizz.buzz.number.NumberFactory;
+
+import java.lang.reflect.InvocationTargetException;
+
+public abstract class AbstractNumberFactory implements NumberProcessor, NumberFactory {
```

Comment:
Good points. How about renaming it to something like `ChainedNumberFactory`? Suggestions for better names are highly appreciated!